### PR TITLE
[stable/redis] Add --no-auth-warning to health checks

### PR DESCRIPTION
**Pull request by bbenzikry**
#### What this PR does / why we need it:
Health checks currently show the warning message
```Warning: Using a password with '-a' or '-u' option on the command line interface may not be safe``` which can be dismissed using the --no-auth-warning flag as mentioned in antirez/redis#5073

Continuation of #15062 
Chart version bumped to 8.0.14 pending #15082

#### Checklist
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

**Comment by bbenzikry** [comment]
@cpanato build didn't pass due to redis-lib-values image tagged with 4.0.11, changed to 5.0.5

**Comment by cpanato** [comment]
@bbenzikry please update your PR with latest master and fix the conflict thanks!

**Review by juan131 — APPROVED** [review]
Thanks for the PR! I'm approving your changes!
You might need to rebase your branch from master since we just merged another PR (see #15082)

**Comment by bbenzikry** [comment]
@juan131 Looks ok now but there seems to be an issue with the cluster, I'm seeing the same as
https://github.com/helm/charts/pull/15128#issuecomment-506862438

File: stable/redis/ci/redis-lib-values.yaml [hunk 1]
```diff
@@ -4,7 +4,7 @@
 image:
   registry: docker.io
   repository: redis
-  tag: '4.0.11'
+  tag: '5.0.5'
```
Comment (juan131) [inline stable/redis/ci/redis-lib-values.yaml:7]: Please undo this change

**Review by juan131 — CHANGES_REQUESTED** [review]

File: stable/redis/ci/redis-lib-values.yaml (hunk 1 above)
Comment (bbenzikry) [reply stable/redis/ci/redis-lib-values.yaml:7]: Flag is not supported on this version
See https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/helm_charts/15100/pull-charts-e2e/1144314462543548416/build-log.txt
```
docker://redis-08ug45oq09:Container failed liveness probe.. Container will be killed and recreated.
I0627 19:08:34.757]   Warning  Unhealthy               8m53s (x9 over 9m38s)  kubelet, gke-jenkins-default-pool-51cafeba-f5cd  Liveness probe failed: 
I0627 19:08:34.757] Warning: Using a password with '-a' option on the command line interface may not be safe.
I0627 19:08:34.757] Unrecognized option or bad number of args for: '--no-auth-warning'
```

Comment (juan131) [reply stable/redis/ci/redis-lib-values.yaml:7]: hmm.. You're right! Skip this comment!

**Review by juan131 — APPROVED** [review]
